web/src/stream_list.ts

```diff
@@ -863,19 +863,35 @@ const update_streams_for_search = _.throttle(actually_update_streams_for_search,
 
 // Exported for tests only.
 export function initialize_stream_cursor(): void {
-    stream_cursor = new ListCursor({
+    stream_cursor = new ListCursor<StreamListRow>({
         list: {
             scroll_container_selector: "#left_sidebar_scroll_container",
             find_li(opts) {
-                const stream_id = opts.key;
-                const $li = get_stream_li(stream_id);
-                return $li;
+                if (opts.key.type === "stream") {
+                    const $li = get_stream_li(opts.key.stream_id);
+                    return $li;
+                }
+                return $(
+                    `#stream-list-${opts.key.section_id}-container .stream-list-toggle-inactive-channels`,
+                );
```

Comment:
This would be better written with a switch on `key.type` with a `default: throw new Error`. 

The main flaw is that having an unknown type result in our assuming it is an inactive channels elements is bad. A few reasons:
- It will silently break if a new type is added.
- Searching/grepping for the inactive-channel type won't find this code.
- The code is less readable, since it's implicitly asserting that a `opts.key` is a `StreamRow`. As a side note, why does `opts` not have a type annotation?

Reply:
`opts` doesn't need a type annotation, it's implied by `ListCursor`